pkg/kp/consulutil/watch.go

```diff
@@ -35,35 +38,59 @@ func WatchPrefix(
 	if pause < 250*time.Millisecond {
 		pause = 250 * time.Millisecond
 	}
+
+	listLatencyHistogram, consulLatencyHistogram, outputPairsBlocking, outputPairsHistogram := watchHistograms("prefix")
+
+	var (
+		safeListStart    time.Time
+		outputPairsStart time.Time
+	)
 	for {
 		select {
 		case <-done:
 			return
 		case <-timer.C:
 		}
-		timer.Reset(pause) // upper bound on request rate
+		safeListStart = time.Now()
 		pairs, queryMeta, err := SafeList(clientKV, done, prefix, &api.QueryOptions{
 			WaitIndex: currentIndex,
 		})
+		listLatencyHistogram.Update(int64(time.Since(safeListStart) / time.Millisecond))
+		if queryMeta != nil {
+			consulLatencyHistogram.Update(int64(queryMeta.RequestTime))
+		}
+		timer.Reset(pause)
 		switch err {
 		case CanceledError:
 			return
 		case nil:
 			currentIndex = queryMeta.LastIndex
+			outputPairsStart = time.Now()
 			select {
 			case <-done:
 			case outPairs <- pairs:
 			}
+			outputPairsBlocking.Update(int64(time.Since(outputPairsStart) / time.Millisecond))
+			outputPairsHistogram.Update(int64(unsafe.Sizeof(pairs)))
```

Comment:
https://play.golang.org/p/YveicAxnTr seems to tell us we'll get the same thing all the time